src/Symfony/Component/Messenger/DependencyInjection/MessengerPass.php

```diff
@@ -206,11 +206,11 @@ private function guessHandledClasses(\ReflectionClass $handlerClass, string $ser
             throw new RuntimeException(sprintf('Invalid handler service "%s": class "%s" must have an "__invoke()" method.', $serviceId, $handlerClass->getName()));
         }
 
-        $parameters = $method->getParameters();
-        if (1 !== \count($parameters)) {
-            throw new RuntimeException(sprintf('Invalid handler service "%s": method "%s::__invoke()" must have exactly one argument corresponding to the message it handles.', $serviceId, $handlerClass->getName()));
+        if (0 === $method->getNumberOfRequiredParameters()) {
+            throw new RuntimeException(sprintf('Invalid handler service "%s": method "%s::__invoke()" first argument must correspond to the message it handles.', $serviceId, $handlerClass->getName()));
```

Comment:
What does "must correspond" mean in this context? Since we don't check that the type matches, shouldn't we just tell that the method must handle at least one argument: the message it handles?

Reply:
@nicholasruunu Any comment on the last comment from @nicolas-grekas?

Reply:
Yes, I have updated the exception message.